### PR TITLE
Hoisted scopes fix

### PR DESCRIPTION
**Pull request by tmat**
The CDI in Windows PDBs is encoded as edge-inclusive, but some of the reader helpers didn't adjust for that. EE did. This change unifies the interpretation and removes a duplicate type that represents the scopes.

### CODE REVIEW

**Comment by tmat** [comment]
test windows_debug_vs-integration_prtest please

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/Core/Portable/PEWriter/Members.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using System.Reflection.Metadata;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.Debugging;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/PEWriter/Members.cs]: Is this change needed?

File: src/Compilers/Core/Portable/PEWriter/CustomDebugInfoWriter.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.Diagnostics;
 using System.Linq;
 using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/PEWriter/CustomDebugInfoWriter.cs]: Is this change needed?

**Comment by AlekseyTs** [comment]
I am curious why VB isn't affected by the issue?

File: src/Compilers/Core/Portable/PEWriter/Members.cs (hunk 1 above)
Comment (tmat) [inline src/Compilers/Core/Portable/PEWriter/Members.cs]: Yes. The StateMachineHoistedLocalScope is in this namespace, the other equivalent type that used to be in Cci namespace is deleted.

File: src/Compilers/Core/Portable/PEWriter/CustomDebugInfoWriter.cs (hunk 2 above)
Comment (tmat) [inline src/Compilers/Core/Portable/PEWriter/CustomDebugInfoWriter.cs]: ditto.

**Comment by tmat** [comment]
VB uses a different encoding for the scope information.

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Ping for second review. Thanks!

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Ping for second review.

**Comment by tmat** [comment]
Thanks Neal
